lib/ruby_smb/dcerpc/srvsvc.rb

```diff
@@ -0,0 +1,96 @@
+module RubySMB
+  module Dcerpc
+    module Srvsvc
+
+      OPNUMS = {
+          net_share_enum_all: 15
+      }
+
+      #https://msdn.microsoft.com/en-us/library/cc247293.aspx
+      class NetShareEnumAll < BinData::Record
+
+        endian :little
+
+        uint32 :referent_id, initial_value: 0x00000001
+        uint32 :max_count,    initial_value: -> { (server_unc.do_num_bytes / 2) + 1 }
+        uint32 :offset,       initial_value: 0
+        uint32 :actual_count, initial_value: -> {max_count}
+        string :server_unc,   pad_front: false,
+               initial_value: -> {host.encode('utf-16le')}
+
+        uint16 :padding, initial_value: 0
+        uint32 :level, initial_value: 1
+
+        uint32 :ctr, initial_value: 1
+        uint32 :ctr_referent_id, initial_value: 0x00000001
+        uint32 :ctr_count, initial_value: 0
+        uint32 :pointer_to_array, initial_value: 0
+
+        uint32 :max_buffer, initial_value: 4294967295
+
+        uint32 :resume_referent_id, value: 0x00000001
+        uint32 :resume_handle, initial_value: 0
+
+        def self.parse_response(response)
```

Comment:
I am wondering if it is possible to handle this with only BinData structures and avoid slicing and unpacking the directly the byte stream?

Reply:
same as above^